File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        pars = {
            '}': '{',
            ')': '(',
            ']': '[',
        }

        stack = list()
        
        for i in s:
            print(stack)
            if i not in pars.keys():
                stack.append(i)
            else:
                if not stack or pars.get(i) != stack.pop():
                    return False

        if  not stack:
            return True
        else:
            return False 

File: Data Structures & Algorithms/minimum-stack/submission-2.cpp
#include <list>
class MinStack {
public:
    MinStack() {
    
    }
    
    void push(int val) {
        _stack.push_back(val);
        if (_history.empty())
            _history.push_back(val); 
        else _history.push_back(min(val, _history.back()));
    }
    
    void pop() {
        _stack.pop_back();
        _history.pop_back();

    }
    
    int top() {
        return _stack.back();
        
    }
    
    int getMin() {
        return _history.back();
    }
private:
    list<int> _stack;
    list<int> _history;
    int _min = 0;
};
